src/backend_helper.c

```diff
@@ -619,18 +635,256 @@ int get_all_options(PrinterCUPS *p, Option **options)
             opts[optsIndex].default_value = get_string_copy("NA");
         }
 
```

Comment:
This can happen for any media size, not only A4. So check at first whether you find the default size in the list of available sizes, if it is not there, convert the name to PPD/PostScript style, IPP style, and legacy, until you find it in the list. Use the functions of libcups (`cups/pwg-media.c`).